core/src/HIP/Kokkos_HIP_Instance.cpp

```diff
@@ -258,6 +258,28 @@ Kokkos::HIP::size_type *HIPInternal::scratch_functor(
   return m_scratchFunctor;
 }
 
+Kokkos::HIP::size_type *HIPInternal::scratch_functor_host(
+    const std::size_t size) const {
+  if (verify_is_initialized("scratch_functor_host") && m_scratchFunctorSize < size) {
+    m_scratchFunctorSize = size;
+
+    using Record = Kokkos::Impl::SharedAllocationRecord<Kokkos::HIPHostPinnedSpace, void>;
+
+    if (m_scratchFunctorHost)
+      Record::decrement(Record::get_record(m_scratchFunctorHost));
+
+    Record *const r =
+        Record::allocate(Kokkos::HIPHostPinnedSpace(), "Kokkos::InternalScratchFunctorHost",
+                         m_scratchFunctorSize);
+
+    Record::increment(r);
+
+    m_scratchFunctorHost = reinterpret_cast<size_type *>(r->data());
+  }
+
+  return m_scratchFunctor;
```

Comment:
```suggestion
  return m_scratchFunctorHost;
```

Reply:
haha this is why the other issue didn't crash the code. `m_scratchFunctorHost` was always null lol, but this thing made it not crash ...